plugins/roleplay.py

```diff
@@ -0,0 +1,317 @@
+        :param msg:
+        :return:
+        """
+
+        args = msg.content.split(" ", 1)
+        if len(args) < 2:
+            raise CommandSyntaxError("Needs an argument, dumbass.")
+        d_pattern = re.compile("(\d+|)d(\d+)(\+\d+|-\d+|)(a|d|)")
+        d_data = d_pattern.search(args[1].lower())
+        if d_data:
+            if d_data.group(1):
+                t_num = min(max(int(d_data.group(1)), 1), 10000)
+            else:
+                t_num = 1
+            t_die = min(max(int(d_data.group(2)), 2), 10000)
+            if d_data.group(3):
+                t_bonus = int(d_data.group(3))
+                t_b = f" with a {t_bonus} modifier"
+            else:
+                t_bonus = 0
+                t_b = ""
+            t_adv = d_data.group(4)
+            if t_adv == 'a':
+                t_s = "an advantageous"
+                t_1 = t_2 = 0
+                t_s1 = t_s2 = ""
+                for i in range(t_num):
+                    t_r = randint(1, t_die)
+                    t_1 += t_r
+                    t_s1 += f"[{t_r}] "
+                    t_r = randint(1, t_die)
+                    t_2 += t_r
+                    t_s2 += f"[{t_r}] "
+                if t_1 > t_2:
+                    t_res = t_1 + t_bonus
+                    t_r_s = t_s1
+                else:
+                    t_res = t_2 + t_bonus
+                    t_r_s = t_s2
+            elif t_adv == 'd':
+                t_s = "a disadvantageous"
+                t_1 = t_2 = 0
+                t_s1 = t_s2 = ""
+                for i in range(t_num):
+                    t_r = randint(1, t_die)
+                    t_1 += t_r
+                    t_s1 += f"[{t_r}] "
+                    t_r = randint(1, t_die)
+                    t_2 += t_r
+                    t_s2 += f"[{t_r}] "
+                if t_1 < t_2:
+                    t_res = t_1 + t_bonus
+                    t_r_s = t_s1
+                else:
+                    t_res = t_2 + t_bonus
+                    t_r_s = t_s2
+            else:
+                t_s = "a"
+                t_res = 0
+                t_r_s = ""
+                for i in range(t_num):
+                    t_r = randint(1, t_die)
+                    t_res += t_r
+                    t_r_s += f"[{t_r}] "
+
+            if t_num > 1:
+                t_r_s = f"\n**ANALYSIS: Rolled dice:** `{t_r_s}`"
+            else:
+                t_r_s = ""
+
+            await respond(msg,
+                          f"**ANALYSIS: {msg.author.display_name} has attempted {t_s} {t_num}D{t_die} roll{t_b}, "
+                          f"getting {t_res}.**{t_r_s}")
+
+    @Command("racerole",
+             doc="Adds or removes roles from the list of race roles that are searched by the bio command.",
+             syntax="add/remove (role mentions)",
+             perms={"manage_messages"},
+             category="role_play")
+    async def _racerole(self, msg):
```

Comment:
While we're implementing this here, I suppose the role picker thing would probably go in here as well?

Reply:
What do you mean by "role picker"?

Reply:
Like SAIL has, the command for receiving a race role

Reply:
Done